src/main/java/org/truffleruby/language/TruffleBootNodes.java

```diff
@@ -88,52 +89,119 @@ public DynamicObject forceContext() {
         }
     }
 
-    @CoreMethod(names = "main", onSingleton = true, optional = 1)
+    @CoreMethod(names = "main", onSingleton = true)
     public abstract static class MainNode extends CoreMethodArrayArgumentsNode {
 
+        @Child private SnippetNode findSFileSnippetNode;
+        @Child private SnippetNode checkSyntaxSnippetNode;
```

Comment:
Yeah, could be, but the snippets are not always used which makes this tiny better. I don't think we would gain anything by this change. I think it's good either way so I keep it as is.